sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/DoubleBuffer.cs

```diff
@@ -13,19 +13,14 @@ namespace Azure.Monitor.OpenTelemetry.LiveMetrics.Internals
     /// The 'FlipDocumentBuffers' method swaps the current buffer with a new one, allowing the
     /// consumer to process the documents in the returned buffer without interference from ongoing writes.
     /// </summary>
-    internal class DoubleBuffer
+    internal class DoubleBuffer<T> where T : class
```

Comment:
Why do we need a templated class here. Do you have plans to use it on different type of objects. If we plan to do this on different objects it may not be a perfect design.